api/rosetta/errors.go

```diff
@@ -296,15 +296,140 @@ func invalidKey(fail failure.InvalidKey) Error {
+func unpackError(err error) *echo.HTTPError {
+	return echo.NewHTTPError(http.StatusBadRequest, invalidEncoding(invalidJSON, err))
+}
+
+// unpackError returns the HTTP status code and Rosetta Error for requests
+// that did not pass validation.
+func formatError(err error) *echo.HTTPError {
+
+	var ibErr failure.InvalidBlockHash
+	if errors.As(err, &ibErr) {
+		return echo.NewHTTPError(http.StatusBadRequest, invalidFormat(ibErr.Description.Text,
+			withDetail("want_length", ibErr.WantLength),
+		))
 	}
-	if errors.Is(err, ErrAddressLength) {
-		return echo.NewHTTPError(http.StatusBadRequest, invalidFormat(AddressLength, withDetail("want_length", HexAddressSize)))
+	var iaErr failure.InvalidAccountAddress
+	if errors.As(err, &iaErr) {
+		return echo.NewHTTPError(http.StatusBadRequest, invalidFormat(iaErr.Description.Text,
+			withDetail("want_length", iaErr.WantLength),
+		))
 	}
-	if errors.Is(err, ErrTxLength) {
-		return echo.NewHTTPError(http.StatusBadRequest, invalidFormat(TxLength, withDetail("want_length", HexIDSize)))
+	var itErr failure.InvalidTransactionHash
+	if errors.As(err, &itErr) {
+		return echo.NewHTTPError(http.StatusBadRequest, invalidFormat(itErr.Description.Text,
+			withDetail("want_length", itErr.WantLength),
+		))
 	}
+	var icErr failure.IncompleteBlock
+	if errors.As(err, &icErr) {
+		return echo.NewHTTPError(http.StatusBadRequest, invalidFormat(icErr.Description.Text))
+	}
+
 	return echo.NewHTTPError(http.StatusBadRequest, invalidFormat(err.Error()))
 }
+
+// apiError returns the HTTP status code and Rosetta Error for various errors
+// occurred during request processing.
+func apiError(description string, err error) *echo.HTTPError {
+
+	var inErr failure.InvalidNetwork
+	if errors.As(err, &inErr) {
+		return echo.NewHTTPError(http.StatusUnprocessableEntity, invalidNetwork(inErr))
+	}
+	var ibErr failure.InvalidBlock
+	if errors.As(err, &ibErr) {
+		return echo.NewHTTPError(http.StatusUnprocessableEntity, invalidBlock(ibErr))
+	}
+	var ubErr failure.UnknownBlock
+	if errors.As(err, &ubErr) {
+		return echo.NewHTTPError(http.StatusUnprocessableEntity, unknownBlock(ubErr))
+	}
+	var iaErr failure.InvalidAccount
+	if errors.As(err, &iaErr) {
+		return echo.NewHTTPError(http.StatusUnprocessableEntity, invalidAccount(iaErr))
+	}
+	var icErr failure.InvalidCurrency
+	if errors.As(err, &icErr) {
+		return echo.NewHTTPError(http.StatusUnprocessableEntity, invalidCurrency(icErr))
+	}
+	var ucErr failure.UnknownCurrency
+	if errors.As(err, &ucErr) {
+		return echo.NewHTTPError(http.StatusUnprocessableEntity, unknownCurrency(ucErr))
+	}
+	var itErr failure.InvalidTransaction
+	if errors.As(err, &itErr) {
+		return echo.NewHTTPError(http.StatusUnprocessableEntity, invalidTransaction(itErr))
+	}
+	var utErr failure.UnknownTransaction
+	if errors.As(err, &utErr) {
+		return echo.NewHTTPError(http.StatusUnprocessableEntity, unknownTransaction(utErr))
+	}
+
+	// Construction API specific errors.
```

Comment:
There should be no space below this if it's a comment for the section underneath. Also, if you do put a comment like this, you should probably also put one on the other section.